### PR TITLE
Cleans up TPM format one error messages.

### PR DESCRIPTION
**Pull request by Superhepper**
- Creates proper sentences for TPM format one error messages. and adds tests for TpmFormatOneResponseCode display trait implementation.

- Changes display trait implementation for ArgumentNumber and adds tests for the enum.

### CODE REVIEW

File: tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_one_argument_number_tests.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        //              P |---Error nr---|
+        // Only bits 6 to 11 is parsed by ArgumentNumber.
+        // P bit is set, indicating parameter argument.
+        let return_code_argument_info: u16 = nr | (TPM2_RC_P as u16);
+        let actual = u8::try_from(
+            return_code_argument_info
+                .checked_shr(6).expect("Failed to extract argument information.")
+            )
+            .map (ArgumentNumber::from)
+            .unwrap_or_else(|_| panic!("Failed to convert argument information {:#b} to u8.", return_code_argument_info));
```
Comment (wiktor-k) [inline tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_one_argument_number_tests.rs:50]: :thinking: I'm wondering if `.unwrap_or_else(|_| panic!(` can just be folded into `expect` here?

File: tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_one_argument_number_tests.rs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        // Only bits 6 to 11 is parsed by ArgumentNumber.
+        // Session bit is clear and P bit is clear, indicating handle argument.
+        let return_code_argument_info: u16 = nr | (TPM2_RC_H as u16);
+        let actual = u8::try_from(
+            return_code_argument_info
+                .checked_shr(6)
+                .expect("Failed to extract argument information."),
+        )
+        .map(ArgumentNumber::from)
+        .unwrap_or_else(|_| {
```
Comment (wiktor-k) [inline tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_one_argument_number_tests.rs:117]: Ditto.

**Review by wiktor-k — APPROVED** [review]
Looks great, thanks!

**Review by ionut-arm — APPROVED** [review]
LGTM!

File: tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_one_argument_number_tests.rs (hunk 1 above)
Comment (Superhepper) [reply tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_one_argument_number_tests.rs:50]: If you write ```.expect(format!())``` you get a lint error. https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call

File: tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_one_argument_number_tests.rs (hunk 2 above)
Comment (Superhepper) [reply tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_one_argument_number_tests.rs:117]: https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call

File: tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_one_argument_number_tests.rs (hunk 1 above)
Comment (wiktor-k) [reply tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_one_argument_number_tests.rs:50]: Ah, right. Thanks for the link and the explanation! 👋

**Pull request merged**